catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/filter/FilterReader.java

```diff
@@ -28,15 +31,20 @@ public FilterReader() throws JAXBException {
     this.context = JAXBContext.newInstance(FilterType.class);
   }
 
-  public JAXBElement<FilterType> unmarshalFilter(InputStream inputStream) throws JAXBException {
+  public JAXBElement<FilterType> unmarshalFilter(InputStream inputStream)
+      throws JAXBException, XMLStreamException {
     return unmarshal(inputStream, FilterType.class);
   }
 
   @SuppressWarnings("unchecked")
   private <T> JAXBElement<T> unmarshal(InputStream inputStream, Class<T> tClass)
-      throws JAXBException {
+      throws JAXBException, XMLStreamException {
+    XMLInputFactory xif = XMLInputFactory.newFactory();
```

Comment:
❓ can't the factory be created once per object rather than in every unmarshall call?

Reply:
Looking at the API for the factory, it doesn't mention anything about thread safety.  https://docs.oracle.com/javase/7/docs/api/javax/xml/stream/XMLInputFactory.html.  I'm not sure we should move it to a global.  It looks like we don't do this elsewhere : https://github.com/codice/ddf/search?utf8=%E2%9C%93&q=XMLInputFactory.newFactory%28%29&type=